operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -75,18 +76,59 @@ type Image struct {
 type RedpandaReconciler struct {
 	// KubeConfig is the [rest.Config] that provides the go helm chart
 	// Kubernetes access. It should be the same config used to create client.
-	KubeConfig    *rest.Config
-	Client        client.Client
-	Scheme        *runtime.Scheme
-	EventRecorder kuberecorder.EventRecorder
-	ClientFactory internalclient.ClientFactory
+	KubeConfig           *rest.Config
+	Client               client.Client
+	Scheme               *runtime.Scheme
+	EventRecorder        kuberecorder.EventRecorder
+	ClientFactory        internalclient.ClientFactory
+	CloudSecretsExpander *pkgsecrets.CloudExpander
+	// CloudSecretsFlags holds the settings required to instantiate a CloudExpander.
+	// This is used to generate the CLI arguments to the bootstrap templater.
+	CloudSecretsFlags CloudSecretsFlags
```

Comment:
I thought we discussed only using the CRD field for this? As is cloud would have to set this on both the operator chart and CRD (And ensure they stay in sync). Could we instead extract these values from the CR during reconciliation?

Reply:
We could. I began by following the v1 approach, though. Cloud doesn't have to double-set this currently: there's code to default the CR values to "what the operator is using" so there shouldn't need to be any stuttering here. Let's have a quick face-to-face next week - I agree this needs sorting out, but perhaps both v1 and v2 can be done at the same time with a follow-up PR.